### PR TITLE
KHP3-788 Display a VMMC Theatre Register

### PR DESCRIPTION
**Pull request by CynthiaKamau**
new img pic

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/cohort/definition/evaluator/VMMCRegisterCohortDefinitionEvaluator.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) throws EvaluationException {
+
+		VMMCRegisterCohortDefinition cohortddef = (VMMCRegisterCohortDefinition) cohortDefinition;
+
+		Cohort newCohort = new Cohort();
+
+		Date startDate = (Date) context.getParameterValue("startDate");
+		Date endDate = (Date) context.getParameterValue("endDate");
+
+		String qry = "SELECT t.patient_id from kenyaemr_etl.etl_vmmc_enrolment t inner join kenyaemr_etl.etl_patient_demographics d on t.patient_id = d.patient_id and date(t.visit_date) BETWEEN date(:startDate) AND date(:endDate) GROUP BY t.patient_id;";
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/cohort/definition/evaluator/VMMCRegisterCohortDefinitionEvaluator.java:55]: Is there a possibility to have multiple enrollments for a patient within a given reporting period?

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCCircumcisionDateDataEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @Autowired
+    private EvaluationService evaluationService;
+
+    public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
+        EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
+
+        String qry = "select patient_id, date(visit_date) from kenyaemr_etl.etl_vmmc_circumcision_procedure GROUP BY patient_id;";
+
+        SqlQueryBuilder queryBuilder = new SqlQueryBuilder();
+        Date startDate = (Date)context.getParameterValue("startDate");
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCCircumcisionDateDataEvaluator.java:41]: you haven't used the params in your queries. No need to have them in the evaluator. Cascade to all evaluators

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCLatestPopulationTypeDataEvaluator.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Map;
+import java.util.Date;
+
+/**
+ * Evaluates Population type
+ */
+@Handler(supports= VMMCLatestPopulationTypeDataDefinition.class, order=50)
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCLatestPopulationTypeDataEvaluator.java:29]: No need to create a new population type. Please reuse the one you used in cacx screening register

**Review by ojwanganto — APPROVED** [review]

**Review by patryllus — COMMENTED** [review]
This should be an encounter based report for obs

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCStartTimeDataEvaluator.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+@Handler(supports=VMMCStartTimeDataDefinition.class, order=50)
+public class VMMCStartTimeDataEvaluator implements PersonDataEvaluator {
+
+    @Autowired
+    private EvaluationService evaluationService;
+
+    public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
+        EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
+
+        String qry = "select patient_id, date(time_of_first_placement_cut) from kenyaemr_etl.etl_vmmc_circumcision_procedure GROUP BY patient_id;";
```
[the remark this answers is not in the document]
Comment (patryllus) [reply api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCStartTimeDataEvaluator.java]: You can use the etl_adverse events table to populate procedure and post procedure Adverse events

[the remark this answers is not in the document]
Comment (patryllus) [reply api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/vmmc/VMMCStartTimeDataEvaluator.java]: For etl_adverse events table use the appropriate form to populate procedure and post procedure Adverse events

**Review by patryllus — COMMENTED** [review]
Looks okay

**Pull request merged**